### PR TITLE
Solved: Create an issue in action repo to let Actions owner know

### PR DESCRIPTION
**Pull request by h0x0er**
Closes step-security/secure-workflows#398

### CODE REVIEW

File: kbanalysis/src/index.ts [hunk 1]
```diff
@@ -1,156 +1,200 @@
...
+
+        const comment_resp = await client.rest.issues.getComment({owner:repos.owner, repo:repos.repo, issue_number:issue_number, comment_id:comment_id})
+        const comment_body = comment_resp.data.body
+            
+        if(comment_body.indexOf(`${target_owner}/${target_main_repo}`) !== -1){
+            // issue already created for repo
+            core.info(`Issue for ${target_owner}/${target_main_repo} is already created.\nExiting`)
+            exit(0)
+        }
+        const content = readFileSync(`knowledge-base/${target_owner.toLocaleLowerCase()}/${target_repo.toLocaleLowerCase()}/action-security.yml`)
```
Comment (varunsh-coder) [inline kbanalysis/src/index.ts:51]: Will this work for nested actions? this part should also have some error handling, such that if content is empty, that a issue is not created.

File: kbanalysis/src/index.ts [hunk 2]
```diff
@@ -1,156 +1,200 @@
...
-            const langs = await client.rest.repos.listLanguages({owner:target_owner, repo:target_repo})
-            lang = Object.keys(langs.data)[0] // top language used in repo
-        }catch(err){
-            lang = "NOT_FOUND"
+    if(resp.data.state === "closed"){
+        if(isPaused(resp.data.labels)){
+            core.info(`Issue creation is paused for ${target_owner}/${target_repo}`)
+            exit(0)
         }
+        const issue_number = 71 // Data Store Issue ID
```
Comment (varunsh-coder) [inline kbanalysis/src/index.ts:39]: is this issue number for this repo or was for your own repo? for this repo, I think the issue is #460.

**Comment by varunsh-coder** [comment]
@h0x0er  what about this as the text?

```
At https://github.com/step-security/secure-workflows we are building a knowledge-base (KB) of GITHUB_TOKEN permissions needed by different GitHub Actions. When developers try to set minimum token permissions for their workflows, they can use this knowledge-base instead of trying to research permissions needed by each GitHub Action they use.

Below you can see the KB of your GitHub Action.

If you think this information is not accurate, or if in the future your GitHub Action starts using a different set of permissions, please create an issue at https://github.com/step-security/secure-workflows/issues to let us know. 

This issue is automatically created by our analysis bot, feel free to close after reading :)

Reference:
GitHub asks users to define workflow permissions, see https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/ and https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token for securing GitHub workflows against supply-chain attacks.

Setting minimum token permissions is also checked for by Open Source Security Foundation (OpenSSF) [Scorecards](https://github.com/ossf/scorecard). Scorecards recommend using https://github.com/step-security/secure-workflows so developers can fix this issue in an easier manner.
```

**Comment by h0x0er** [comment]
@varunsh-coder Looks good..

**Pull request merged**